### ISSUE TITLE
Support <const> and <close> attributes in local variables

### ISSUE THREAD

**Issue by pavelxdd** [pigpigyyy/Yuescript#97]
I can define local variable via `local` keyword, but yuescript doesn't support attributes from lua 5.4

This works:
````lua
local a = 123
````

This doesn't work
````lua
local a <const> = 123
local a <close> = 123
````

My project has mutex userdata which should be marked as to-be-closed variable to automatically unlock when it goes out of scope.

The only solution I see now in using macro function to write raw lua code.

**Comment by pigpigyyy** [comment]
Sorry for forgetting to mention it in the doc. You can now use attributes from Lua in Yuescript.
```moonscript
const a = 1
close _ = close#: ->
```
get
```lua
local a <const> = 1 -- 1
local _ <close> = setmetatable({ }, { -- 2
  __close = function() end -- 2
}) -- 2
```

**Comment by pavelxdd** [comment]
These lines:
````moonscript
x = 1 if y ~= 2
local x = 1 if y ~= 2
````

generate the following code:
````lua
local x -- 1
if y ~= 2 then -- 1
  x = 1 -- 1
end -- 1
````

but these lines:
````moonscript
const x = 1 if y ~= 2
close x = 1 if y ~= 2
````

generate code with local variable `x` inside the `if` block:

````lua
if y ~= 2 then -- 1
  local x <const> = 1 -- 1
end -- 1
````

**Comment by pigpigyyy** [comment]
Confirmed that the attribute syntax with line decorator compiles to codes of inconsistent behavior.
```moonscript
const a = 1 if true
```
Should compile to
```lua
local a <const> = (function()
  if true then
    return 1
  end
end)()
-- instead of
if true then
  local a <const> = 1
end
```

**Comment by pigpigyyy** [comment]
The attribute syntax with line decorator issue should now be fixed by commit a9fe6a5c4c17cf7e8305c1d614939ce510fb1103.

**Comment by pavelxdd** [comment]
Thanks, it works
